### PR TITLE
Since Gutenberg 4.5 Block scripts need to be registered in the footer

### PR DESCRIPTION
**Pull request by imath**
Hi Riad,

I was testing your plugin and found this JavaScript error:
`Uncaught TypeError: Cannot read property 'registerBlockType' of undefined`

As I had the same issue with my own blocks and resolved it making sure JavaScript was registered into the footer, I thought you'd be interesting to update/fix this very usefull plugin too 😉 

👋

### CODE REVIEW

File: wp-js-plugin-starter.php [hunk 1]
```diff
@@ -31,7 +31,9 @@ function wp_js_plugin_starter_register_block() {
 	wp_register_script(
 		'wp-js-plugin-starter',
 		wp_js_plugin_starter_url( 'dist/index.js' ),
-		array( 'wp-element' )
+		array( 'wp-element' ),
+		'1.0.1',
+		true
```
Comment (youknowriad) [inline wp-js-plugin-starter.php]: I don't think that's the way to fix the issue. The real issue is that if you use `wp.editor` for instance, you need to add `wp-editor` as a dependency etc...

Comment (imath) [reply wp-js-plugin-starter.php]: Thanks! Then, for this specific case, I suggest to change the dependency to `wp-blocks`

File: wp-js-plugin-starter.php [hunk 2]
```diff
@@ -31,11 +31,12 @@ function wp_js_plugin_starter_register_block() {
 	wp_register_script(
 		'wp-js-plugin-starter',
 		wp_js_plugin_starter_url( 'dist/index.js' ),
-		array( 'wp-element' )
+		array( 'wp-blocks' ),
```
Comment (youknowriad) [inline wp-js-plugin-starter.php]: Thanks for the update, I think we need both `wp-element` and `wp-blocks` here. Also as you may know, the commits in this particular repository have an importance. So I might have to rebase your commit post merge (which means you won't be visible as a contributor) :(

**Comment by imath** [comment]
From what I saw ˋwp-element` is a dependency of ˋwp-blocks`.

Sure no problem about not having me referenced into the commit.

I’m happy to contribute to this great tool.

**Comment by youknowriad** [comment]
Even if `wp-element` is a dependency of `wp-blocks` we shouldn't assume it is. Because if `wp-blocks` is changed at some point and drop this dependency while wee still use it, it may cause issues. My rule is: if you use something directly, add it as a dependency.

File: wp-js-plugin-starter.php (hunk 2 above)
Comment (youknowriad) [inline wp-js-plugin-starter.php]:
```suggestion
		array( 'wp-blocks', 'wp-element' ),
```

Comment (imath) [reply wp-js-plugin-starter.php]: Great, thanks a lot.

**Pull request merged**